src/main/java/com/spotify/confidence/eventsender/EventSenderEngineImpl.java

```diff
@@ -43,8 +44,7 @@ public void run() {
 
         if (flushPolicies.stream().anyMatch(FlushPolicy::shouldFlush)) {
           flushPolicies.forEach(FlushPolicy::reset);
-          eventStorage.createBatch();
-          uploadQueue.add(UPLOAD_SIG);
+          uploadQueue.add(new UploadMsg(UPLOAD_SIG, eventStorage.createBatch()));
```

Comment:
If the `UploadPoller` thread is slow for any reason, we might enqueue multiple "UploadMsg"s that will contain (partly) the same batches in it, leading to event duplications. Am I understanding this correctly?